tests/test_basic.c

```diff
@@ -0,0 +1,237 @@
+
+  amqp_channel_open_ok_t *res =
+      amqp_channel_open(connection_state_, fixed_channel_id);
+  assert(res != NULL);
+
+  return connection_state_;
+}
+
+void close_channel_and_connection(amqp_connection_state_t connection_state_) {
+  amqp_rpc_reply_t rpc_reply =
+      amqp_channel_close(connection_state_, 1, AMQP_REPLY_SUCCESS);
+  assert(rpc_reply.reply_type == AMQP_RESPONSE_NORMAL);
+
+  rpc_reply = amqp_connection_close(connection_state_, AMQP_REPLY_SUCCESS);
+  assert(rpc_reply.reply_type == AMQP_RESPONSE_NORMAL);
+
+  int rc = amqp_destroy_connection(connection_state_);
+  assert(rc == AMQP_STATUS_OK);
+}
+
+void basic_publish(amqp_connection_state_t *connectionState_,
+                   const char *message_) {
+  amqp_bytes_t message_bytes;
+  message_bytes.len = strlen(message_);
+  message_bytes.bytes = (char *)message_;
+
+  amqp_basic_properties_t properties;
+  properties._flags = 0;
+
+  properties._flags |= AMQP_BASIC_DELIVERY_MODE_FLAG;
+  properties.delivery_mode = AMQP_DELIVERY_NONPERSISTENT;
+
+  int retval = amqp_basic_publish(
+      *connectionState_, fixed_channel_id, amqp_cstring_bytes(""),
+      amqp_cstring_bytes(test_queue_name),
+      /* mandatory=*/1,
+      /* immediate=*/0,  // RabbitMQ 3.x does not support the "immediate" flag
+                         // according to
+                         // https://www.rabbitmq.com/specification.html
+      &properties, message_bytes);
+
+  assert(retval == 0);
+}
+
+void queue_declare(amqp_connection_state_t *connection_state_,
+                   const char *queue_name_) {
+  amqp_queue_declare_ok_t *res = amqp_queue_declare(
+      *connection_state_, fixed_channel_id, amqp_cstring_bytes(queue_name_),
+      /*passive*/ 0,
+      /*durable*/ 0,
+      /*exclusive*/ 0,
+      /*auto_delete*/ 1, amqp_empty_table);
+  assert(res != NULL);
+}
+
+void receive_body_frames(amqp_connection_state_t *connection_state_,
+                         char *body_, uint64_t body_size_) {
+  char *body_begin = body_;
+  uint64_t body_read, body_left;
+  for (body_read = 0, body_left = body_size_; body_read < body_size_;) {
+    amqp_frame_t body_frame;
+    int rc = amqp_simple_wait_frame(*connection_state_, &body_frame);
+    assert(rc == AMQP_STATUS_OK);
+
+    assert(body_frame.frame_type == AMQP_FRAME_BODY);
+
+    amqp_bytes_t *body_fragment = &body_frame.payload.body_fragment;
+
+    assert(body_fragment->len <= body_left);
+
+    memcpy(body_begin + body_read, body_fragment->bytes, body_fragment->len);
+    body_read += body_fragment->len;
+    body_left -= body_fragment->len;
+  }
+}
+
+char *basic_get(amqp_connection_state_t *connection_state_,
+                const char *queue_name_, uint64_t *out_body_size_) {
+  amqp_rpc_reply_t rpc_reply;
+  do {
+    rpc_reply = amqp_basic_get(*connection_state_, fixed_channel_id,
+                               amqp_cstring_bytes(queue_name_), /*no_ack*/ 1);
+  } while (rpc_reply.reply_type == AMQP_RESPONSE_NORMAL &&
```

Comment:
Probably shouldn't loop forever in case the message doesn't land in a queue.  Instead have a limit to the number of times that it loops, or a limit in time.

Reply:
The ctest runner already limits the time the test may take, doesn't it?

Reply:
Not by default (a flag needs to be passed to ctest).  Travis will timeout the build after something like 10 or 20 minutes.